conda_package_streaming/transmute.py

```diff
@@ -89,14 +77,42 @@ def transmute(
                 else:
                     tar_get.addfile(member)
 
-            pkg_tar.close()
-            pkg_stream.close()
-
             info_tar.close()
-            info_stream.close()
+            pkg_tar.close()
 
-        with conda_file.open(f"info-{file_id}.tar.zst", "w") as info_file:
-            info_file.write(info_io.getvalue())
+            info_size = info_file.tell()
+            pkg_size = pkg_file.tell()
+
+            info_file.seek(0)
+            pkg_file.seek(0)
+
+        with zipfile.ZipFile(
+            os.path.join(path, f"{file_id}.conda"),
+            "x",  # x to not append to existing
+            compresslevel=zipfile.ZIP_STORED,
+        ) as conda_file:
+
+            # Use a maximum of one Zstd compressor, stream_writer at a time to save memory.
+            data_compress = compressor()
+
+            with conda_file.open(
+                f"pkg-{file_id}.tar.zst", "w"
+            ) as pkg_file_zip, data_compress.stream_writer(
+                pkg_file_zip, size=pkg_size, closefd=False
+            ) as pkg_stream:
+                pkg_dat = pkg_file._file.read()
+                pkg_stream.write(pkg_dat)
```

Comment:
Remind me, does this trigger reading the whole file into memory?

```suggestion
                pkg_stream.write(pkg_file._file.read())
```

Reply:
We probably need copyfileobj.